[Go 1.12] Local build broken due to #2252

Running `./util/docker-dev/dev_data.sh` returns:

2020/11/18 22:40:47 populate_dev_data.go:54: Adding dev data to local emulator...
2020/11/18 22:40:47 populate_dev_data.go:145: Adding local (empty) secrets...
2020/11/18 22:40:47 appengine.go:102: Closing clients
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x98552c]

goroutine 1 [running]:
cloud.google.com/go/datastore.(*Client).PutMulti(0x0, 0xd53dc0, 0xc0000bc010, 0xc000295120, 0x1, 0x1, 0xaeab20, 0xc0003f6520, 0x0, 0x0, ...)
	/home/user/go/pkg/mod/cloud.google.com/go/datastore@v1.3.0/datastore.go:516 +0x17c
cloud.google.com/go/datastore.(*Client).Put(0x0, 0xd53dc0, 0xc0000bc010, 0xc0003b7fc0, 0xace420, 0xc0003bd5a0, 0x0, 0x0, 0x42d271)
	/home/user/go/pkg/mod/cloud.google.com/go/datastore@v1.3.0/datastore.go:490 +0xe8
github.com/web-platform-tests/wpt.fyi/shared.cloudDatastore.Put(0xd53dc0, 0xc0000bc010, 0x0, 0xd4f6c0, 0xc0003b7fc0, 0xace420, 0xc0003bd5a0, 0xc000295208, 0xa69b1c, 0xbdfdc0, ...)
	/home/user/wpt.fyi/shared/datastore_cloud.go:152 +0x14e
main.addSecretToken(0xd5f1e0, 0xc0003f6500, 0xc52a67, 0xc, 0xace420, 0xc0003bd5a0)
	/home/user/wpt.fyi/util/populate_dev_data.go:324 +0xc0
main.main()
	/home/user/wpt.fyi/util/populate_dev_data.go:147 +0x1109
exit status 2
make: *** [dev_data] Error 1
Makefile:236: recipe for target 'dev_data' failed

This is caused by `datastore: missing project/dataset id` in [appengine.go](https://github.com/web-platform-tests/wpt.fyi/blob/master/shared/appengine.go#L49).

In addition, when running `./util/docker-dev/web_server.sh`, I am seeing error messages:

goroutine 1 [running]:
main.main()
	/home/user/wpt.fyi/webapp/web/main.go:33 +0xe2
ERROR    2020-11-18 22:20:54,201 instance.py:284] Cannot connect to the instance on localhost:19939
ERROR    2020-11-18 22:20:54,525 instance.py:284] Cannot connect to the instance on localhost:20760
ERROR    2020-11-18 22:20:54,846 instance.py:284] Cannot connect to the instance on localhost:20721
2020/11/18 22:20:56 Closing clients
panic: dialing: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.

goroutine 1 [running]:
main.main()
	/home/user/wpt.fyi/webapp/web/main.go:33 +0xe2
2020/11/18 22:20:57 Closing clients
panic: dialing: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.

--- a/Makefile
+++ b/Makefile
@@ -110,7 +110,7 @@ web_components_test: xvfb firefox chrome webapp_node_modules_all psmisc
 lighthouse: chrome webapp_node_modules_all
 	cd webapp; npx lhci autorun --failOnUploadFailure
 
-dev_appserver_deps: gcloud-app-engine-python gcloud-app-engine-go gcloud-cloud-datastore-emulator pip-grpcio
+dev_appserver_deps: gcloud-app-engine-python gcloud-app-engine-go gcloud-cloud-datastore-emulator java pip-grpcio
 
 chrome: wget
 	# Pinned to Chrome 84 to workaround https://github.com/web-platform-tests/wpt.fyi/issues/2128

--- a/api/interop.go
+++ b/api/interop.go
@@ -121,10 +121,9 @@ func loadFallbackInteropRun(ctx context.Context, filters shared.TestRunFilter) (
 	}
 
 	// Iterate until we find interop data where its TestRunIDs match the query.
-	var interop metrics.PassRateMetadataLegacy
 	it := query.Run(store)
-	found := false
 	for {
+		var interop metrics.PassRateMetadataLegacy
 		_, err := it.Next(&interop)
 		if err == store.Done() {
 			return nil, nil
@@ -134,13 +133,10 @@ func loadFallbackInteropRun(ctx context.Context, filters shared.TestRunFilter) (
 		if keysChecker != nil && !keysChecker(interop.TestRunIDs) {
 			continue
 		}
-		found = true
+		result = &interop
 		break
 	}
-	if !found {
-		return nil, nil
-	}
-	return &interop, nil
+	return result, nil
 }
 
 func checkKeysAreAligned(shaKeys map[string]shared.KeysByProduct) func(shared.TestRunIDs) bool {

--- a/util/docker-dev/dev_data.sh
+++ b/util/docker-dev/dev_data.sh
@@ -4,4 +4,4 @@ DOCKER_DIR=$(dirname $0)
 source "${DOCKER_DIR}/../commands.sh"
 
 # Run util/populate_dev_data.go (via make) in the docker environment.
-wptd_exec make dev_data FLAGS=\"$@\"
+wptd_exec make dev_data FLAGS=\"-project=wptdashboard-local -remote_host=staging.wpt.fyi -datastore_host=localhost:8001\"

--- a/util/docker-dev/web_server.sh
+++ b/util/docker-dev/web_server.sh
@@ -48,6 +48,9 @@ wptd_exec_it dev_appserver.py \
    --admin_port=8000 \
    --api_host=$WPTD_CONTAINER_HOST \
    --api_port=9999 \
-   -A=wptdashboard \
+   --support_datastore_emulator=true \
+   --datastore_consistency_policy=consistent \
+   --datastore_emulator_port=8001 \
+   -A=wptdashboard-local \
    /home/user/wpt.fyi/webapp/web/app.dev.yaml
 

